File: src/Java/QuickSort.java
package Java;

public class QuickSort {
    static public int[] quickSort(int[] numeros) {
        quickSort(numeros, 0, numeros.length - 1);
        return numeros;
    }

    static private void quickSort(int[] numeros, int min, int max) {
        if (min >= max) {
            return;
        }
        int pivotIndex = min + (max - min) / 2;
        int pivot = numeros[pivotIndex];
        int i = min, j = max;
        while (i <= j) {
            while (numeros[i] < pivot) {
                i++;
            }
            while (numeros[j] > pivot) {
                j--;
            }
            if (i <= j) {
                int temp = numeros[i];
                numeros[i] = numeros[j];
                numeros[j] = temp;
                i++;
                j--;
            }
        }
        if (min < j) {
            quickSort(numeros, min, j);
        }
        if (max > i) {
            quickSort(numeros, i, max);
        }
    }
}

File: src/Kotlin/QuickSort.kt
package Kotlin

fun quickSort(arr: IntArray, bottom: Int, top: Int) {
    if (bottom < top) {
        val pivotIndex = partition(arr, bottom, top)

        quickSort(arr, pivotIndex + 1, top)
        quickSort(arr, bottom, pivotIndex - 1)
    }


}

fun partition(arr: IntArray, bottom: Int, top: Int): Int {
    val pivot = arr[top]
    var i = bottom - 1

    for (j in bottom until top) {
        if (arr[j] <= pivot) {
            i++

            val temp = arr[i]
            arr[i] = arr[j]
            arr[j] = temp
        }

        val temp = arr[i + 1]
        arr[i + 1] = arr[top]
        arr[top] = temp

    }
    return i + 1
}

fun main() {
    val arr = intArrayOf(64, 34, 25, 12, 22, 11, 90)
    println("Unsorted array: ${arr.joinToString()}")

    quickSort(arr, 0, arr.size - 1)

    println("Sorted array: ${arr.joinToString()}")
}





